File: mddProject/src/app/feature/themes/interface/theme.ts
import { User } from "../../me/interface/user";

export interface Theme {
    id: number;                    // Identifiant du thème
    title: string;                 // Titre du thème
    description: string;           // Description du thème
    subscribedUsers: User[];       // Liste des utilisateurs abonnés à ce thème
  }

File: back/src/main/java/com/mdd/pocmdd/models/Theme.java
package com.mdd.pocmdd.models;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "themes")
public class Theme {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToMany(mappedBy = "subscribedThemes")
    private List<User> subscribedUsers = new ArrayList<>();

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, length = 255)
    private String description;

}
